src/rgw/rgw_lc.cc

```
@@ -1249,32 +1252,243 @@ class LCOpAction_Transition : public LCOpAction {
+
+    attrs.erase(RGW_ATTR_ID_TAG);
+    attrs.erase(RGW_ATTR_TAIL_TAG);
+
+    obj_op.write_meta(tier_ctx.o.meta.size, 0, attrs, null_yield);
+    if (r < 0) {
+      return r;
+    }
+
+    return 0;
+  }
+
+  int transition_obj_to_cloud(lc_op_ctx& oc) {
+    std::shared_ptr<RGWRESTConn> conn;
+
+    /* init */
+    string id = "cloudid";
+    string endpoint=oc.tier.endpoint; 
+    RGWAccessKey key = oc.tier.key;
+    HostStyle host_style = oc.tier.host_style;
+    string bucket_name = oc.tier.target_path;
+    const RGWZoneGroup& zonegroup = oc.store->svc()->zone->get_zonegroup();
+   
+    if (bucket_name.empty()) {
+      bucket_name = "rgwx-" + zonegroup.get_name() + "-" + oc.tier.storage_class +
+                    "-cloud-bucket";
+      boost::algorithm::to_lower(bucket_name);
+    }
+
+    conn.reset(new S3RESTConn(oc.cct, oc.store->svc()->zone,
+                                id, { endpoint }, key, host_style));
+
+    /* http_mngr */
+    RGWCoroutinesManager crs(oc.store->ctx(), oc.store->getRados()->get_cr_registry());
+    RGWHTTPManager http_manager(oc.store->ctx(), crs.get_completion_mgr());
+
+    int ret = http_manager.start();
+    if (ret < 0) {
+      ldpp_dout(oc.dpp, 0) << "failed in http_manager.start() ret=" << ret << dendl;
+      return ret;
+    }
+
+    RGWLCCloudTierCtx tier_ctx(oc.cct, oc.o, oc.store, oc.bucket->get_info(),
+                        oc.obj->get_obj(), oc.rctx, conn, bucket_name, oc.tier.target_storage_class,
+                        &http_manager);
+    tier_ctx.acl_mappings = oc.tier.acl_mappings;
+    tier_ctx.multipart_min_part_size = oc.tier.multipart_min_part_size;
+    tier_ctx.multipart_sync_threshold = oc.tier.multipart_sync_threshold;
+    tier_ctx.storage_class = oc.tier.storage_class;
+
+    bool al_tiered = false;
+    ret = crs.run(new RGWLCCloudCheckCR(tier_ctx, &al_tiered));
+    
+    if (ret < 0) {
+      ldpp_dout(oc.dpp, 0) << "ERROR: failed in RGWCloudCheckCR() ret=" << ret << dendl;
```

not returning here on purpose?

---

yes.. "RGWLCCloudCheckCR" verifies if the object is already transitioned to remote cloud (incase of multisite). Since its just best effort to avoid duplicate transition, I thought it may be fine to ignore any error here.

---

Can you add a comment explaining that?

---

sure..done.